src/Ucommerce.Transactions.Payments.QuickpayLink/QuickpayServiceClient.cs

```diff
@@ -0,0 +1,266 @@
+
+            var paymentDto = DeseralizeContent<PaymentModel>(response.Content);
+            state = paymentDto.State;
+            return paymentDto.Operations.Any(x => x.Type == PaymentCanceled);
+        }
+
+        /// <summary>
+        /// Refunds the payment.
+        /// </summary>
+        /// <param name="payment"></param>
+        /// <param name="state"></param>
+        /// <returns></returns>
+        public bool RefundPayment(Payment payment, out string state)
+        {
+            var formContent = new FormUrlEncodedContent(new[]
+            {
+                new KeyValuePair<string, string>("amount", Convert.ToInt32(payment.Amount.ToCents()).ToString()),
+            });
+
+            var response = Task.Run(async () => await CreateWebRequest($"payments/{payment.TransactionId}/refund", Method.POST, formContent))
+                   .ConfigureAwait(false)
+                   .GetAwaiter()
+                   .GetResult();
+
+            if (response.StatusCode != HttpStatusCode.Accepted)
+            {
+                state = "Bad request";
+                return false;
+            }
+
+            var paymentDto = DeseralizeContent<PaymentModel>(response.Content);
+            state = paymentDto.State;
+            return paymentDto.Operations.Any(x => x.Type == PaymentRefunded);
+        }
+
+        #region Private methods
+        private async Task<HttpResponseMessage> CreateWebRequest(string url, Method method, HttpContent content = null)
+        {
+            var httpClient = new HttpClient();
+            httpClient.DefaultRequestHeaders.Add("Authorization", $"Basic {Convert.ToBase64String(Encoding.Default.GetBytes($":{_apiKey}"))}");
+            httpClient.DefaultRequestHeaders.Add("accept-version", ApiVersion);
+
+            var uri = $"{BasePath}{url}";
+            HttpResponseMessage reponse;
+            switch (method)
+            {
+                case Method.GET:
+                    reponse = await httpClient.GetAsync(uri);
+                    break;
+
+                case Method.POST:
+                    reponse = await httpClient.PostAsync(uri, content);
+                break;
+
+                case Method.PUT:
+                    reponse = await httpClient.PutAsync(uri, content);
+                    break;
+
+                default:
+                    throw new ArgumentException($"Http Method not implemented: {nameof(method)}.");
+            }
+
+            return reponse;
+        }
+
+        private T DeseralizeContent<T>(HttpContent content)
+        {
+            var contentStr = Task.Run(async () => await content.ReadAsStringAsync())
+                               .ConfigureAwait(false)
+                               .GetAwaiter()
+                               .GetResult();
+
+            return JsonSerializer.Deserialize<T>(contentStr);
+        }
+        #endregion
+    }
+
+    /// <summary>
+    /// Represents the set of http methodss the client supports.
```

Comment:
Might as well, now that we are fixing small things, fix this extra s

```suggestion
    /// Represents the set of http methods the client supports.
```

Reply:
Fixed the typo.